### ISSUE TITLE
PubSub: Subscriber does not propagate exception to main thread:

### ISSUE THREAD

**Issue by jam182** [GoogleCloudPlatform/google-cloud-python#5392]
pubsub v0.34.0

Subscriber type: `subscribe_experimental`

# Issue
The following error usually only happens when trying to send a `ack()` or `nack()`:

```
ERROR:google.cloud.pubsub_v1.subscriber._protocol.helper_threads:ValueError: Can not send() on an RPC that has never been open()ed.
Exception in thread Thread-LeaseMaintainer:
Traceback (most recent call last):
  File "/usr/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python3.6/site-packages/google/cloud/pubsub_v1/subscriber/_protocol/leaser.py", line 152, in maintain_leases
    requests.ModAckRequest(ack_id, p99) for ack_id in ack_ids])
  File "/usr/lib/python3.6/site-packages/google/cloud/pubsub_v1/subscriber/_protocol/dispatcher.py", line 157, in modify_ack_deadline
    self._manager.send(request)
  File "/usr/lib/python3.6/site-packages/google/cloud/pubsub_v1/subscriber/_protocol/streaming_pull_manager.py", line 225, in send
    self._rpc.send(request)
  File "/usr/lib/python3.6/site-packages/google/cloud/pubsub_v1/subscriber/_protocol/bidi.py", line 361, in send
    super(ResumableBidiRpc, self).send, request)
  File "/usr/lib/python3.6/site-packages/google/cloud/pubsub_v1/subscriber/_protocol/bidi.py", line 353, in _recoverable
    raise exc
  File "/usr/lib/python3.6/site-packages/google/cloud/pubsub_v1/subscriber/_protocol/bidi.py", line 348, in _recoverable
    return method(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/google/cloud/pubsub_v1/subscriber/_protocol/bidi.py", line 236, in send
    'Can not send() on an RPC that has never been open()ed.')
ValueError: Can not send() on an RPC that has never been open()ed.
```

Unfortunately the exception is only printed, but not propagated to the `future`, so the child thread dies, while the subscriber looks healthy. But it's indeed not pulling messages anymore.

# Context
When we see this happening, (or around the same time) we also see that the average time for `ack`-ing a message goes up significantly. Might blame this to network congestion, so that might hint to a loss of connection by the `rpc` client that either does not retry to connect or it just can't, and fails doing so. But in such case the appropriate error should be raised and surfaced. In fact even having several instances of the subscriber, when this incident happens, it happens pretty much at the same time across all the instances. A network problem would explain why all instances tend to error at the same time, with the same error.

There is a chance that this issue is similar or somehow related to this other one: https://github.com/GoogleCloudPlatform/google-cloud-python/issues/5306

# Details
python version: 3.6.3
## pubsub packages:
google-api-core==1.1.2
google-auth==1.4.1
google-cloud-pubsub==0.34.0
googleapis-common-protos==1.5.3
grpc-google-iam-v1==0.11.4
grpcio==1.12.0

**Comment by theacodes** [comment]
Hi there. I believe we fixed this in 0.35.0, can you try it with that?

**Comment by tseaver** [comment]
Duplicate of #5306.
